bundles/org.openhab.persistence.jdbc/src/main/java/org/openhab/persistence/jdbc/db/JdbcBaseDAO.java

```diff
@@ -645,6 +643,10 @@ public Map<String, String> getSqlTypes() {
     }
 
     public String getDataType(Item item) {
-        return sqlTypes.get(getItemType(item));
+        String dataType = sqlTypes.get(getItemType(item));
+        if (dataType == null) {
+            throw new UnsupportedOperationException("No data type found for " + getItemType(item));
```

Comment:
Same here, no need to mention the exception `UnsupportedOperationException` in the method prototype ?

Reply:
Ditto.